src/contexts/Townhall.tsx

```diff
@@ -18,6 +18,12 @@ interface Props {
 
 export const TownhallContext = React.createContext<Townhall | null>(null);
 
+type Dispatch = React.Dispatch<SetStateAction<Townhall | undefined >>  | null;
```

Comment:
I'm not sure that inside the SetStateAction we should have `Townhall | undefined` I think it should just be Townhall and undefined is outside of the generic like the null.

Reply:
I'm wrong!